tests/test_config.py

```diff
@@ -182,3 +182,23 @@ def test_prompt_encoder_warning_num_layers(self):
             PromptEncoder(config)
         expected_msg = "for MLP, the argument `encoder_num_layers` is ignored. Exactly 2 MLP layers are used."
         assert str(record.list[0].message) == expected_msg
+
+    def test_ia3_is_feedforward_subset(self):
```

Comment:
Thanks for adding this test. I would like to see a few small amendments:

1. The valid config here uses str and as such is not really tested. Could you please add a test with a valid config that uses lists? Ideally, the 3 test cases can be split into separate functions (testing a single thing per test is generally preferred, even if currently peft is not doing such a good job at that).
2. You can simplify the `assertRaises` call, as it accepts a `msg`/`regex` argument.

Reply:
1. My initial thinking was that valid configs with lists were tested in any case in all the other IA3 tests. I think you're right that it's better to test all valid and invalid configs in one place. so i added one test for lists here, and refactored a bit. 
2. Looks like `assertRaises` doesn't have a `regex` argument, and `msg` is the failure message in case the test fails ( and not the error message when the error is raised). I think you meant `assertRaisesRegex`. I switched to this, should be cleaner now! Also, currently I'm doing an exact match for the error message. Might be too strict if you decide to reword the message later on, and I don't know what's best practice here.